### PR TITLE
add usage tracking in the api token

### PR DESCRIPTION
**Pull request by Elhebert**
### What does it do?

Add telemetry to the API token

### CODE REVIEW

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js [hunk 1]
```diff
@@ -37,6 +38,12 @@ const ApiTokenCreateView = () => {
   const { lockApp, unlockApp } = useOverlayBlocker();
   const toggleNotification = useNotification();
   const history = useHistory();
+  const { trackUsage } = useTracking();
+  const trackUsageRef = useRef(trackUsage);
+
+  useEffect(() => {
+    trackUsageRef.current('didEditTokenFromList');
+  }, []);
```
Comment (ronronscelestes) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js:46]: after testing it, I think it calls trackUsage with didEditTokenFromList even if the user is creating.

maybe changing it this way would fix it:
`  useEffect(() => {
     if(!isCreating) {
       trackUsageRef.current('didEditTokenFromList');
     }
  }, []);
`

or maybe we could event move the logic directly into the UpdateButton like this?
`
    <LinkUpdate
      to='PATH'
      title='TITLE'
      onClick={() => trackUsageRef('didEditTokenFromList')}
    >
      <Pencil />
    </LinkUpdate>
`

**Review by ronronscelestes — CHANGES_REQUESTED** [review]
I added a small comment/question about didEditTokenFromList but apart from that good job 🤗

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js (hunk 1 above)
Comment (Elhebert) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js:46]: Based on the list of events the `didEditTokenFromList` is supposed to be triggered:
> When the user lands on the create or edit token page 

But I we could definitely add two new events:
- `willCreateTokenFromList`
- `didCreateTokenFromList`

That way we have different events for the creation and update action done from the ListView. 

WDYT?

Ping @JabStrapi.

Comment (ronronscelestes) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js:46]: Oh right I didn't know that, thanks for the explanation!
Indeed having willCreateTokenFromList and didCreateTokenFromList as separate events could be interesting :)

Comment (JabStrapi) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js]: Let's add two new events yes! But I'd recommend to go for `willAddTokenFromList` and `didAddTokenFromList`.

**Review by ronronscelestes — APPROVED** [review]
LGTM :rocket:

**Pull request merged**